File: packages/types/src/movement.ts
import type { Currency } from "./account";

export type MovementType = "income" | "expense" | "transfer";

export interface Movement {
  id: string;
  type: MovementType;
  amount: number;
  currency: Currency;
  description: string;
  date: string;
  accountId: string;
  destinationAccountId?: string;
  categoryId: string;
  exchangeRate?: number;
  destinationAmount?: number;
  createdAt: string;
  updatedAt: string;
}

export interface CreateMovementInput {
  type: MovementType;
  amount: number;
  description: string;
  date: string;
  accountId: string;
  destinationAccountId?: string;
  categoryId: string;
  exchangeRate?: number;
}

export interface MovementFilters {
  accountId?: string;
  categoryId?: string;
  type?: MovementType;
  dateFrom?: string;
  dateTo?: string;
  minAmount?: number;
  maxAmount?: number;
}

File: apps/api/app/schemas/movement.py
from pydantic import BaseModel


class CreateMovementInput(BaseModel):
    type: str  # income | expense | transfer
    amount: float
    description: str
    date: str
    accountId: str
    destinationAccountId: str | None = None
    categoryId: str
    exchangeRate: float | None = None


class UpdateMovementInput(BaseModel):
    type: str | None = None
    amount: float | None = None
    description: str | None = None
    date: str | None = None
    accountId: str | None = None
    destinationAccountId: str | None = None
    categoryId: str | None = None


class MovementResponse(BaseModel):
    id: str
    type: str
    amount: float
    currency: str
    description: str
    date: str
    accountId: str
    destinationAccountId: str | None = None
    categoryId: str
    exchangeRate: float | None = None
    destinationAmount: float | None = None
    createdAt: str
    updatedAt: str


class MonthlySummary(BaseModel):
    income: float
    expense: float
    net: float


class PaginatedMovements(BaseModel):
    data: list[MovementResponse]
    total: int
    page: int
    pageSize: int
